src/parachain/issue.ts

```diff
@@ -207,17 +203,30 @@ export class DefaultIssueAPI extends DefaultTransactionAPI implements IssueAPI {
 
     async request(
         amount: MonetaryAmount<WrappedCurrency, BitcoinUnit>,
-        vaultId?: AccountId,
+        vaultAccountId?: AccountId,
+        collateralCurrencyIdLiteral?: CurrencyIdLiteral,
         atomic: boolean = true,
         retries: number = 0,
-        cachedVaults?: Map<AccountId, MonetaryAmount<WrappedCurrency, BitcoinUnit>>
+        cachedVaults?: Map<string, MonetaryAmount<WrappedCurrency, BitcoinUnit>>,
```

Comment:
I'm also wondering about the logic of this function. If a user has a vault selected the user would still need to send the cached vaults as well to prevent the function from loading those again.

Maybe the `cachedVaults` should be `selectedVaults` and if there is only a single element then this is the vault the user wants to issue with? Otherwise the `cachedVaults` and the `vaultAccountId` together with the `collateralCurrencyIdLiteral` are somewhat redundant arguments.

Reply:
If a user specifies a `vaultAccountId`, the function returns without fetching any vaults. If none is specified, the `cachedVaults` array is used to select vaults for the issue request. In the absence of `cachedVaults`, the vault list is queried from the parachain. I think there's no redundance? As above, `vaultAccountId` and `collateralCurrencyIdLiteral` could be merged into `InterbtcPrimitivesVaultId`, which also includes the wrapped currency.